mitodl/ol-data-platform
# create staging and intermediate models for OCW courses

### What are the relevant tickets?




https://github.com/mitodl/ol-data-platform/issues/493
and https://github.com/mitodl/hq/issues/3609

### Description (What does it do?)

Adds the OCW courses metadata related tables into data platform
 - stg__ocw__studio__postgres__websites_website
 - stg__ocw__studio__postgres__websites_websitecontent
 - stg__ocw__studio__postgres__websites_websitestarter
 - int__ocw__courses  (this table can be used for https://github.com/mitodl/hq/issues/3609 e.g. https://bi.odl.mit.edu/queries/1353)
 - int__ocw__course_topics (supporting table for OCW course topic in tabular format)
 - int__ocw__course_instructors (supporting table for OCW instructors metadata in tabular format)
 - int__ocw__course_departments (supporting table for OCW department number and name in tabular format)


### How can this be tested?

Run the following commands, tests should all pass
```
dbt build --select staging.ocw  --vars 'schema_suffix: <YOURNAME>' --target dev_production
dbt build --select intermediate.ocw --vars 'schema_suffix: <YOURNAME>' --target dev_production
```

Run query in https://mitol.galaxy.starburst.io/query-editor to see the result of the above tables. e.g.
```
SELECT * FROM ol_data_lake_production.ol_warehouse_production_rlougee_intermediate.int__ocw__course_topics
```
### Additional Context


- I use https://github.com/mitodl/ocw-hugo-themes/blob/main/base-theme/data/departments.json to map department number to name since it's not maintained in database
- A few old courses have duplicated topics. e.g. https://ocw.mit.edu/courses/3-a26-freshman-seminar-the-nature-of-engineering-fall-2005/data.json. I deduplicate in `int__ocw__course_topics`
